src/olympia/versions/models.py

```diff
@@ -1423,6 +1438,103 @@ def __str__(self):
+            self.ANDROID_LIMITED_COMPATIBILITY_RANGE[1]
+        )
+        is_range_bigger_than_limited_range = self.min.version_int < version_int(
+            self.ANDROID_LIMITED_COMPATIBILITY_RANGE[0]
+        ) and self.max.version_int >= version_int(
+            self.ANDROID_LIMITED_COMPATIBILITY_RANGE[1]
+        )
+        return (
+            is_min_in_limited_range
+            or is_max_in_limited_range
+            or is_range_bigger_than_limited_range
+        )
+
+    def save(self, *args, **kwargs):
+        if (
+            self.application == amo.ANDROID.id
+            and self.version.addon.type == amo.ADDON_EXTENSION
+        ):
+            # Regardless of whether the compat information came from the
+            # manifest, on Android we currently only allow non-promoted/line
+            # extensions to be compatible with Fennec or Fenix that has general
+            # availability, so we might need to override the compat data before
+            # saving it.
+            if self.version_range_contains_forbidden_compatibility():
+                # Arbitrarily pick a range that makes sense and is valid.
+                self.min = AppVersion.objects.filter(
+                    application=amo.ANDROID.id,
+                    version=amo.MIN_VERSION_FENIX_GENERAL_AVAILABILITY,
+                ).first()
+                self.max = AppVersion.objects.filter(
+                    application=amo.ANDROID.id,
+                    version=amo.DEFAULT_WEBEXT_MAX_VERSION,
+                ).first()
+
+            # In addition we need to set/clear strict compatibility.
+            # If the range max is below the first Fenix version, then we need
+            # to set strict_compatibility to True on the File to prevent that
+            # version from being installed in Fenix. If it's not, then we need
+            # to clear it if it was set.
+            #
+            # Strict compatibility also affects Desktop, so there is a slight
+            # edge case if the developer had also set a max on desktop without
+            # expecting it to be obeyed, but that should be fairly rare.
+            if self.max.version_int < version_int(amo.MIN_VERSION_FENIX):
+                strict_compatibility = True
+            else:
+                strict_compatibility = False
+            if self.version.file.strict_compatibility != strict_compatibility:
+                self.version.file.update(strict_compatibility=strict_compatibility)
+
+        rval = super().save(*args, **kwargs)
+        if hasattr(self.version, '_compatible_apps'):
+            del self.version._compatible_apps
+        return rval
+
+    def delete(self, *args, **kwargs):
+        if (
+            self.application == amo.ANDROID.id
+            and self.version.addon.type == amo.ADDON_EXTENSION
+            and self.version.file.strict_compatibility
+        ):
+            self.version.file.update(strict_compatibility=False)
```

Comment:
There's an edge case here where `strict_compatibility` was `True` anyway (we didn't override it) but now we're setting it wrongly to `False`.

Reply:
(That we probably don't care about, just calling it out)

Reply:
Yeah. But there shouldn't be any. The only extension in prod with `strict_compatibility` enabled is one of ours, an old hotfix add-on from 2019 that is currently disabled. Was the comment I put not enough ? I can reword it if needed.

Reply:
The comment was in a different function (save), but if it's in reality an edge case that doesn't happen it doesn't matter anyway.